src/Attachment.php

```
@@ -263,7 +263,7 @@ public function save(
         }
 
         /** @var resource $fp */
-        if ($fp = fopen($attachment_path, 'w')) {
+        if ($fp = @fopen($attachment_path, 'w')) {
```

If we're handling this error, we have to silence the standard warning, else it comes down to this test failure:
```
1) Tests\PhpMimeMailParser\ExceptionTest::testSaveAttachmentsWithoutPermissions
fopen(/tmp/attach_m0001iqHSBu/attach01): failed to open stream: Permission denied
```

---

I'm agree with all the proposition except this one. 

I don't know if I'm right, but it seems weird for me to add the error control operators in my src code.
And I think I read that it will be removed in the next version of PHP.

What do you think?

---

This is totally wrong if you're not going to handle this error. But this is not the case. Very much opposite. You're not ignoring any errors, but throwing an exception instead of a warning. 

For example, [here you can see](https://github.com/symfony/filesystem/blob/master/Filesystem.php#L53) Symfony developers do the same thing with @ and an exception. And [here's another case](https://github.com/thephpleague/flysystem/blob/87f1b8155bc78b514c76af79c6d14405df0a6f97/src/Adapter/Local.php#L103) for another major dependency of Laravel. This is a very common pattern when handling these warnings. 

Hopefully you can see that `@` isn't evil per se, but what is more important is why you're using it. In this particular case we're using it to make errors more apparent by throwing an exception. And judging from examples this is the way to do it, how do you think?

---

Ok great improvements. Thanks